### ISSUE TITLE
Rundeck + Ansible always send --limit to playbooks

### ISSUE THREAD

**Issue by javierurien** [rundeck-plugins/ansible-plugin#102]
I have a playbook that looks like this:
```
- hosts: localhost
  vars:
    some_var: 'some value' 
  tasks:
    - name: some local task
      set_fact:
        some_local_var: 'some local value'
- hosts: linux
  tasks:
    - name: some linux task
      shell: 'uname -a'
- hosts: windows
  tasks:
    - name: some windows task
      raw: 'ping 127.0.0.1'
```

I run it from command line with:
`# ansible-playbook some_playbook.yml`
And works as expected.

But I cant find a way to run it from rundeck with ansible plugin. (rundeck shell command works)

What I see is that rundeck runs ansible-playbook with the --limit option always and this playbook needs to be run as shown.

My questions are:
Am I writing bad playbooks?
Is there a way to run ansible playbooks without sending a --limit option?

Thanks

**Comment by jarosite** [comment]
+ 1 I am using playbooks, and this limit for me sometimes are useless. because all node assignment is in yml file.

**Comment by javierurien** [comment]
I think this is due to the way rundeck models running processes.
It has to always have a target.

Perhaps, the ansible plugin could interpret that running on local node means no target (or --limit)

**Comment by brokenjacobs** [comment]
Not just this. But specifying --limit localhost causes ansible to interpret 'localhost' differently inside playbooks. There is an 'implicit' localhost included in inventories that is removed when a limit expression is passed to ansible-playbook. My understanding is that this is done for compatibility with ansible-pull mode. 

For instance I have a playbook that connects to vmware and creates vm's, from 'localhost'. But it breaks in rundeck with -l localhost. So I have to do all sorts of things to make it work with the limit expression:
- Add 'localhost' to my python dynamic inventory with ansible_connection=local set as a hostvar
- Set --connection=local on the command line to ansible-playbook (otherwise ansible will ssh)
- use delegate_to in each step of the playbook
- set ansible_python_interpreter as an extra var (since python interpreter processing kicks in as if this is a 'remote' host. (I'm using a python virtualenv for my rundeck user.)

Basically lots of stuff changes when --limit localhost is used. It forces ansible to treat 'localhost' as remote.

Additionally I can't add this host to inventory inside the playbook and then go configure it, because the --limit localhost will override any limits in the playbook.

**The 'easy' fix to all this would be to allow a field to be set in the plugin to NOT pass the --limit flag to ansible-playbook when 'Execute Locally' is selected for the Node specification. In fact this should probably be the default behavior, as --limit localhost breaks things substantially inside playbooks making it different from expected behavior.**

**Comment by brokenjacobs** [comment]
This appears to be a simple change to the code. Would a Pull request be accepted for this change?

**Comment by frozenice** [comment]
@brokenjacobs of course, as stated at the bottom of the README "Discussions and pull requests are welcome."!
[Rundeck 2.9.0](https://github.com/rundeck/rundeck/issues/2577) will have exciting new features that we need to look into.
I'm currently trying to put some time into this project again (working on a Docker image).

**Comment by DerekTBrown** [comment]
I am running in to this problem, so I thought I would comment here for documentation.  Here are the offending lines:

```
    String limit = nodes.length() > 0 ? nodes.substring(0, nodes.length() - 1): "";
    if (limit != "") {
        configuration.put(AnsibleDescribable.ANSIBLE_LIMIT,limit);
    }
```

**Comment by pixdrift** [comment]
Also just ran into this problem with current version of RunDeck 2.9.2 (2017-08-07). When selecting 'Execute Locally' the Ansible playbook execution is always passed an empty limit parameter resulting in error.

**Comment by DerekTBrown** [comment]
@frozenice is there any harm in removing the `--limit` flag?

**Comment by frozenice** [comment]
@DerekTBrown I don't think so, but am not sure. Needs some testing, I guess. :)

**Comment by larry-s** [comment]
--limit is normally required when sending commands to Ansible as this specifies which nodes are going to be targeted by the commands. As @brokenjacobs says, the --limit and its parameter should be omitted when Nodes is Execute locally.

**Comment by DerekTBrown** [comment]
@larry-s I think that would create a change in the functionality (that isn't necessarily a problem, just something we need to agree upon).  Currently if I execute locally, it will only run against localhost, which might break other peoples' current jobs.

I think there are two possible solutions to this problem:

1. Remove the `--limit` option entirely

**Pros:**
Allows users to pass custom `--limit` using extra arguments.
**Cons:**
Changes existing functionality. 

2. Override `--inventory` with an ad-hoc list of hosts containing only localhost (somehow parse out variables).

**Pros**
Allows users to run `add_host` without error
**Cons**
Doesn't use static inventory file.

**Comment by DerekTBrown** [comment]
I am going to add this as an option as @brokenjacobs suggested.

**Comment by DerekTBrown** [comment]
https://github.com/Batix/rundeck-ansible-plugin/pull/143

**Comment by brokenjacobs** [comment]
Just catching up here. Excellent news!

**Comment by bbomgardner** [comment]
Thanks! Testing this out today in 2.1.2 release. Is it possible to set this option in the job/task or only in Project/Framework?
